server/protocols/jmap/src/main/java/org/apache/james/jmap/event/PropagateLookupRightListener.java

```diff
@@ -110,19 +110,12 @@ private void updateLookupRight(MailboxPath mailboxPath, Stream<Entry> entries, M
 
     private void applyLookupRight(MailboxPath mailboxPath, MailboxSession session, Entry entry) {
         try {
-            boolean hasLookupRight = rightManager.listRights(mailboxPath, session)
-                .getEntries()
-                .getOrDefault(entry.getKey(), MailboxACL.NO_RIGHTS)
-                .contains(Right.Lookup);
-
-            if (!hasLookupRight) {
-                rightManager.applyRightsCommand(mailboxPath,
-                    MailboxACL.command()
-                        .rights(Right.Lookup)
-                        .key(entry.getKey())
-                        .asAddition(),
-                    session);
-            }
+            rightManager.applyRightsCommand(mailboxPath,
+                MailboxACL.command()
+                    .rights(Right.Lookup)
+                    .key(entry.getKey())
+                    .asAddition(),
+                session);
```

Comment:
So you are prefering a design where write is faster than read? Cassandra optimization?
(really I've no strong opinion on this)

Reply:
not really, the mapper can very well optimize for the storage, in cassandra, writing is cheap, in SQL maybe we can have a read-before-write strategy but it's up to the storage layer.